File: challenges/5-cobaias/cpp/raffreitas/cobaias.cpp
#include <bits/stdc++.h>

using namespace std;

void cobaias(int n) {
  int rato = 0, sapo = 0, coelho = 0, quant;
  float total = 0;
  char tipo;

  for (int i = 0; i < n; i++) {
    cin >> quant >> tipo;

    switch (tipo) {
    case 'C':
      coelho += quant;
      break;
    case 'R':
      rato += quant;
      break;
    case 'S':
      sapo += quant;
      break;
    }
    total += quant;
  }

  cout << "Total: " << total << " cobaias" << endl;
  cout << "Total de coelhos: " << coelho <<endl;
  cout << "Total de ratos: " << rato << endl;
  cout << "Total de sapos: " << sapo << endl;
  cout << fixed << setprecision(2) << "Percentual de coelhos: " << (coelho / total) * 100 << " %" <<endl;
  cout << fixed << setprecision(2)  << "Percentual de ratos: " << (rato / total) * 100 << " %" <<endl;
  cout << fixed << setprecision(2)  << "Percentual de sapos: " << (sapo / total) * 100 << " %" <<endl;
 
  return;
}

int main() {
  int n;
  
  cin >> n;

  cobaias(n);

  return 0;
}

File: challenges/5-cobaias/php/akiratorres/cobaias.php
<?php

$coelhos = 0;
$ratos = 0;
$sapos = 0;
$total = 0;

$casos = (int)readline("Digite o número de casos de teste: ");
echo "Digite a quantidade de animais e o tipo conforme o exemplo: 10 C\n\n";

for ($i = 0; $i < $casos; $i++) {
    $caso = explode(' ', readline($i + 1 . "- "));
    $quantia = (int)$caso[0];
    $tipo = $caso[1];

    if (($tipo == 'C') || ($tipo == 'c')) {
        $coelhos += $quantia;
    } else if (($tipo == 'R') || ($tipo == 'r')) {
        $ratos += $quantia;
    } else if (($tipo == 'S') || ($tipo == 's')) {
        $sapos += $quantia;
    }

    $total += $quantia;
}

$percentualCoelhos = ($coelhos * 100) / $total;
$percentualRatos = ($ratos * 100) / $total;
$percentualSapos = ($sapos * 100) / $total;

$percentualCoelhos = number_format($percentualCoelhos, 2, '.', '');
$percentualRatos = number_format($percentualRatos, 2, '.', '');
$percentualSapos = number_format($percentualSapos, 2, '.', '');

echo PHP_EOL . "Total: $total cobaias" . PHP_EOL;
echo "Total de coelhos: $coelhos\nTotal de ratos: $ratos\nTotal de sapos: $sapos" . PHP_EOL;
echo "Percentual de coelhos: $percentualCoelhos %" . PHP_EOL;
echo "Percentual de ratos: $percentualRatos %" . PHP_EOL;
echo "Percentual de sapos: $percentualSapos %" . PHP_EOL;
